sklearn/compose/_target.py

```diff
@@ -75,9 +75,10 @@ class TransformedTargetRegressor(
 
     inverse_func : function, default=None
         Function to apply to the prediction of the regressor. Cannot be set at
-        the same time as `transformer`. The function needs to return a
-        2-dimensional array. The inverse function is used to return
-        predictions to the same space of the original training labels.
+        the same time as `transformer`. The inverse function is used to return
+        predictions to the same space of the original training labels. If
+        `inverse_func` is set, `func` also needs to be provided. The function
+        needs to return a 2-dimensional array.
```

Comment:
Nit: At this point both `func` and inverse func are in context. I think it's better to be explicit:

```suggestion
        `inverse_func` is set, `func` also needs to be provided. The inverse
        function needs to return a 2-dimensional array.
```